wgpu-hal/src/vulkan/instance.rs

```diff
@@ -591,8 +591,10 @@ impl crate::Instance<super::Api> for super::Instance {
 
     unsafe fn create_surface(
         &self,
-        has_handle: &impl raw_window_handle::HasRawWindowHandle,
+        has_handle: &(impl raw_window_handle::HasRawWindowHandle
+              + raw_window_handle::HasRawDisplayHandle),
     ) -> Result<super::Surface, crate::InstanceError> {
+        use raw_window_handle::RawDisplayHandle;
         use raw_window_handle::RawWindowHandle;
 
         match has_handle.raw_window_handle() {
```

Comment:
If we match on `(has_handle.raw_window_handle(), has_handle.raw_display_handle())` we can unpack both halves at once, preventing needing a separate if let inside each branch.

Reply:
This is a concept new to me? How would you match on two enums at the same time?

Reply:
Because you're matching on the tuple of the enums, so you can then in each arm do `(Rwh::Xcb(..), Rdh::Xcb(..))` and it will only match that branch if they're both Xcb (now we know they will always be the same, so this isn't a problem).

Reply:
That is awesome. Did not know you could match tuples like that. I am updating the code RN